### PR TITLE
Modifications in is_int_dtype function

### PR DESCRIPTION
**Pull request by AlexandraDI**
Changed the docstring in is_int_dtype function and removed and replaced a lamda function

### CODE REVIEW

**Comment by AlexandraDI** [comment]
It seems that some tests are not successful. Can you please give me feedback about what I have done wrong?

**Comment by Auxia** [comment]
Hi Alexandra,

I noticed in your examples that you are directly trying to print the result `>>>print(ivy.is_int_dtype(ivy.int32))`

However, the docstrings test requires you to pass the result into a variable and then printing it to console in this manner,
```
>>> x = ivy.is_int_dtype(ivy.int8)
>>> print(x)
True
```

Implementing these changes should resolve the docstring tests.

For the Linting errors, I'd suggest you go through the [pre-commit](https://lets-unify.ai/ivy/contributing/0_setting_up.html#pre-commit) section in the contributing guide and follow the steps to make sure linting errors are resolved automatically when you make a pull request.

**Comment by AlexandraDI** [comment]
The lint error is resolved. I also changed the docstring but the "test-docstring" is still failing. Can you please check again?

**Comment by Auxia** [comment]
It seems there are issues with `str` inputs and a general issue with `ivy.Container` which is causing the docstring test to fail. The examples themselves dont seem to be the issue. I will have to do further testing on this and get back to you.

File: ivy/functional/ivy/data_type.py [hunk 1]
```diff
@@ -692,19 +692,99 @@ def closest_valid_dtype(type: Union[ivy.Dtype, str, None]) -> Union[ivy.Dtype, s
...
+    >>> print(x.dtype)
+    float32
+
+    >>> print(ivy.is_int_dtype(x))
+    False
+
+    With :code:`ivy.NativeArray` input:
+
+    >>> x = ivy.native_array([[-1, -1, -1], [1, 1, 1]], \
+        dtype='int16')
```
Comment (Auxia) [inline ivy/functional/ivy/data_type.py:758]: `dtype = ivy.int16` and not 'int16'

**Comment by Auxia** [comment]
Can you also try to pull from the latest master branch and then committing the requested changes? You can refer to [this](https://lets-unify.ai/ivy/contributing/1_the_basics.html#managing-your-fork) for help in that.

File: ivy/functional/ivy/data_type.py (hunk 1 above)
Comment (AlexandraDI) [reply ivy/functional/ivy/data_type.py]: I changed this but still it is failing :(

**Comment by AlexandraDI** [comment]
> Can you also try to pull from the latest master branch and then committing the requested changes? You can refer to [this](https://lets-unify.ai/ivy/contributing/1_the_basics.html#managing-your-fork) for help in that.

I merged with the upstream master branch.

**Comment by AlexandraDI** [comment]
Unfortunately I still get errors :(

**Comment by AlexandraDI** [comment]
Do you know what I should fix? Is the rest of the function correct?

**Comment by Auxia** [comment]
The tests are passing locally but failing in the CI, it is a weird behaviour but since the examples look good, I will merge it for now.

Thanks for Contributing :)

**Pull request merged**